src/nvim/path.c

```diff
@@ -1761,7 +1761,7 @@ int pathcmp(const char *p, const char *q, int maxlen)
   /* ignore a trailing slash, but not "//" or ":/" */
   if (c2 == NUL
       && i > 0
-      && !after_pathsep((char_u *)s, (char_u *)s + i)
+      && !after_pathsep((char *)s, (char *)s + i)
```

Comment:
Like before, I was trying to control the number of changes I was making - and I wasn't entirely sure that change would be desired, so I didn't want to lump it all together.  

I don't have it finished yet, but I'm working on reworking that to honor the const qualifier, and add that to the PR with the vim_getenv refactor.